File: Backend/RecipeBook/Domain/Entities/Recipe.cs
namespace Domain.Entities;

public class Recipe : Entity
{
    public User Author { get; init; }
    public string Name { get; set; }
    public string Description { get; set; }
    public int CookTime { get; set; }
    public int PortionCount { get; set; }
    public string ImageName { get; set; }

    public ICollection<Tag> Tags { get; set; } = new HashSet<Tag>();
    public ICollection<Step> Steps { get; set; } = new List<Step>();
    public ICollection<Ingredient> Ingredients { get; set; } = new List<Ingredient>();
    public ICollection<Like> Likes { get; set; } = new HashSet<Like>();
    public ICollection<Favorite> Favorites { get; set; } = new HashSet<Favorite>();

    public Recipe( User author, string name, string description, int cookTime, int portionCount, string imageName)
    {
        Author = author;
        Name = name;
        Description = description;
        CookTime = cookTime;
        PortionCount = portionCount;
        ImageName = imageName;
    }

    public Recipe() { }
}

File: Frontend/recipe-book/src/types/recipe.ts
export type Recipe = {
  recipeId: number;
  authorId: number;
  login: string;
  name: string;
  description: string;
  tags: Tag[];
  steps: Step[];
  ingredients: Ingredient[];
  favoritesCount: number;
  isFavorite: boolean;
  likesCount: number;
  isLiked: boolean;
  cookTime: number;
  portionCount: number;
  imageName: string;
};

export type Tag = {
  id: number | undefined;
  name: string;
};

export type Step = {
  id: number | undefined;
  description: string;
};

export type Ingredient = {
  id: number | undefined;
  title: string;
  description: string;
};
